src/GitVersion.Configuration/ConfigurationBuilderBase.cs

```diff
@@ -24,6 +24,7 @@ internal abstract class ConfigurationBuilderBase<TConfigurationBuilder> : IConfi
     private string? commitDateFormat;
     private bool updateBuildNumber;
     private SemanticVersionFormat semanticVersionFormat;
+    private VersionStrategies[] versionStrategies;
```

Comment:
I don't think we should use an array for the `enum`. We should only use an array in the YAML file. There's a bit of relevant discussion in https://github.com/aaubry/YamlDotNet/issues/679. We may have to rely on custom deserialization logic for this specific property. The goal is to make YAML as developer-friendly as possible, without negatively affecting the `GitVersionConfiguration` domain model in C#.

```suggestion
    private VersionStrategies versionStrategies;
```

Reply:
done